### PR TITLE
Use scylla production mode but disable it in CI

### PR DESCRIPTION
**Pull request by jvff**
## Motivation

PR #2603 is failing in CI because our custom runners isn't fully operational.

## Proposal

Tweak the CI workflow to not run in our custom runner.

## Test Plan

CI should pass after this change.

## Release Plan

- Nothing to do / These changes follow the usual release cycle.
- These changes should be backported to the latest `devnet` branch, then
    - be released in a new SDK,
    - be released in a validator hotfix.
- These changes should be backported to the latest `testnet` branch, then
    - be released in a new SDK,
    - be released in a validator hotfix.
- Backporting is not possible but we may want to deploy a new `devnet` and release a new
      SDK soon.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -48,19 +49,26 @@ jobs:
...
+      - name: Setup upterm session
+        uses: lhotari/action-upterm@v1
       - name: Sync
         uses: nick-fields/retry@v2
         with:
           # Docker compose can take some time to start
           max_attempts: 5
           timeout_minutes: 2
-          command: sleep 60 && linera --wallet docker/wallet.json --storage rocksdb:docker/linera.db sync-balance
+          command: sleep 120 && linera --wallet docker/wallet.json --storage rocksdb:docker/linera.db sync-balance
```
Comment (ma2bd) [inline .github/workflows/docker.yml:74]: This one should not be needed.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -48,19 +49,26 @@ jobs:
...
         run: docker build . -f docker/Dockerfile -t linera
+      - name: Patch Dockerfile to run ScyllaDB in developer mode
+        run: sed -i -e '/SCYLLA_AUTO_CONF/{N;s/command:/  SCYLLA_ENABLE_EXPERIMENTAL: 1/p;
+N;N;N;N;d}' docker/docker-compose.yml
       - name: Run Compose
         run: |
           cd docker
           ./compose.sh
+      - name: Setup upterm session
+        uses: lhotari/action-upterm@v1
```
Comment (ma2bd) [inline .github/workflows/docker.yml:67]: I don't think this is needed

File: .github/workflows/docker.yml (hunk 1 above)
Comment (ma2bd) [reply .github/workflows/docker.yml]: Also what is the meaning of `sleep 120` and `timeout_minutes: 2` ?

File: .github/workflows/docker.yml (hunk 2 above)
Comment (ma2bd) [reply .github/workflows/docker.yml]: (although it doesn't hurt)

Comment (ma2bd) [reply .github/workflows/docker.yml]: It does seem to hurt actually

**Review by christos-h — APPROVED** [review]
Nice! I like the commented out steps. Easy to put back in if something goes wrong. But perhaps an indicator as to _why_ they are commented out for future readers?

**Pull request merged**